engines/misc/src/main/java/org/alfresco/transform/misc/transformers/TextToPdfContentTransformer.java

```diff
@@ -344,6 +403,137 @@ public PDDocument createPDFFromText(Reader text, int pageLimit)
             }
             return doc;
         }
+
+        public void setFont(String aFontName)
+        {
+            PDType1Font font = PagedTextToPDF.getStandardFont(aFontName);
+
+            if (font != null)
+            {
+                super.setFont(font);
+            }
+
+            this.fontName = aFontName;
+        }
+
+        private PDFont getFont(PDDocument doc, String name)

```

Comment:
This method is quite complex. I'd suggest either splitting it up (maybe even extracting some logic to a new class) or adding javadoc to describe what it does.

Reply:
Splitted `getFont(PDDocument, String)` method code into 3 new methods, each one with a different approach to get the font:

- `getFromStandardFonts(String fontName)`
- `getFromFontMapper(String, PDDocument)`
- `getFromFileSystem(String)`

Added Javadoc to each method.